KarhooUISDK/Screens/CheckoutScreen/NewCheckout/Workers/KarhooNewCheckoutBookingWorker.swift

```diff
@@ -77,62 +83,199 @@ final class KarhooNewCheckoutBookingWorker: NewCheckoutBookingWorker {
         self.flightNumber = flightNumber
     }
 
+    func update(comment: String?) {
+        self.comment = comment
+    }
+
     func performBooking() {
         switch bookingState {
         case .loading: break
         default:
             bookingState = .loading
             reportBookingEvent()
-            if sdkConfiguration.authenticationMethod().isGuest() {
-                submitGuestBooking()
-            } else {
+            if isKarhooUser() {
                 submitAuthenticatedBooking()
+            } else {
+                submitGuestBooking()
```

Comment:
Let's name this as `submitGuestAndTokenExchangeBooking()` to reflect the full use. Otherwise it may not be so obvious to someone checking the code out what happened to the token exchange flow.